### PR TITLE
[NN] add multihead in DotGatConv

### PR DESCRIPTION
**Pull request by Ericcsr**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by Ericcsr** [comment]
It seems that there is no unit test for DotGatConv in tests/pytorch/test_nn.py Hence the new version of code directly pass the unit test.

**Review by zhjwy9343 — COMMENTED** [review]
Please refine the Description section at the doc stream. Add multiple head formula.

File: python/dgl/nn/pytorch/conv/dotgatconv.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ class DotGatConv(nn.Module):
         same value.
     out_feats : int
         Output feature size; i.e, the number of dimensions of :math:`h_i^{(l+1)}`.
+    num_heads : int
+        Number of head in Multi-Head Attention
```
Comment (sneakerkg) [inline python/dgl/nn/pytorch/conv/dotgatconv.py:45]: Please also change the formula in the doc and verify which dimension the multi-head will applied on.

**Review by sneakerkg — CHANGES_REQUESTED** [review]

File: python/dgl/nn/pytorch/conv/dotgatconv.py (hunk 1 above)
Comment (Ericcsr) [reply python/dgl/nn/pytorch/conv/dotgatconv.py:45]: OK

**Comment by yzh119** [comment]
Why don't we rename this module as Transformer? This is identical to Transformer's (restricted to graph structure) sublayer except for the rescale of attention scores (Transformer divides attention score by $\sqrt{d}$ where $d$ refers to the dimension per head).

**Comment by jermainewang** [comment]
> It seems that there is no unit test for DotGatConv in tests/pytorch/test_nn.py Hence the new version of code directly pass the unit test.

Then please add unittests for it. Thanks!

**Comment by Ericcsr** [comment]
> > It seems that there is no unit test for DotGatConv in tests/pytorch/test_nn.py Hence the new version of code directly pass the unit test.
> 
> Then please add unittests for it. Thanks!

Noticed!

**Review by sneakerkg — APPROVED** [review]

**Pull request merged**